homeassistant/components/generic/config_flow.py

```diff
@@ -283,17 +269,16 @@ async def async_step_user(
                 if not errors:
                     user_input[CONF_CONTENT_TYPE] = still_format
                     user_input[CONF_LIMIT_REFETCH_TO_URL_CHANGE] = False
-                    if user_input.get(CONF_STILL_IMAGE_URL):
-                        await self.async_set_unique_id(self.flow_id)
-                        return self.async_create_entry(
-                            title=name, data={}, options=user_input
-                        )
-                    # If user didn't specify a still image URL,
-                    # we can't (yet) autodetect it from the stream.
-                    # Show a conditional 2nd page to ask them the content type.
-                    self.cached_user_input = user_input
-                    self.cached_title = name
-                    return await self.async_step_content_type()
+                    if still_url is None:
+                        # If user didn't specify a still image URL,
+                        # The automatically generated still image that stream generates
+                        # is always jpeg
+                        user_input[CONF_CONTENT_TYPE] = "image/jpeg"
+
+                    await self.async_set_unique_id(self.flow_id)
```

Comment:
Please remove setting the unique id as there's no unique id.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unique-id-requirements

Reply:
@MartinHjelmare Ok... i tried this, but removing the unique id lines breaks the integration: when I add a new camera it creates an entry in the config ui with no entities.

Maybe you can see what I'm doing wrong?:
https://github.com/davet2001/home-assistant/tree/generic_remove_unique_id

Reply:
Don't use the config entry unique_id as entity unique_id. If there's no proper unique_id for the devices/entities, use the config entry entry_id.

Reply:
https://github.com/home-assistant/core/blob/7b75a16745afebb5e878f03069fa7068da748c63/homeassistant/components/generic/camera.py#L112

Reply:
@MartinHjelmare Done in #70568